include/swift/AST/EvaluatorDependencies.h

```diff
@@ -32,80 +33,13 @@ namespace detail {
-/// further dependency edges.
-///
-/// A cascading edge is always conservatively correct to produce, but it comes
-/// at the cost of increased resources spent (and possibly even wasted!) during
-/// incremental compilation. A private edge, by contrast, is more efficient for
-/// incremental compilation but it is harder to safely use.
-///
-/// To ensure that these edges are registered consistently with the correct
-/// scopes, requests that act as the source of dependency edges are required
-/// to specify a \c DependencyScope under which all evaluated sub-requests will
-/// register their dependency edges. In this way, \c DependencyScope values
-/// form a stack-like structure and are pushed and popped by the evaluator
-/// during the course of request evaluation.
-///
-/// When determining the kind of scope a request should use, always err on the
-/// side of a cascading scope unless there is absolute proof any discovered
-/// dependencies will be private. Inner requests may also defensively choose to
-/// flip the dependency scope from private to cascading in the name of safety.
-enum class DependencyScope : bool {
-  Private = false,
-  Cascading = true,
-};
-
-/// Returns a \c DependencyScope appropriate for the given (formal) access level.
-///
-/// :warning: This function exists to bridge the old manual reference
-/// dependencies code to the new evaluator-based reference dependencies code.
-/// The manual code often made private/cascading scope judgements based on the
-/// access level of a declaration. While this makes some sense intuitively, it
-/// does not necessarily capture an accurate picture of where real incremental
-/// dependencies lie. For example, references to formally private types can
-/// "escape" to contexts that have no reference to the private name if, say,
-/// the layout of that private type is taken into consideration by
-/// SILGen or IRGen in a separate file that references the declaration
-/// transitively. However, due to the density of the current dependency
-/// graph, redundancy in registered dependency edges, and the liberal use of
-/// cascading edges, we may be saved from the worst consequences of this
-/// modelling choice.
-///
-/// The use of access-levels for dependency decisions is an anti-pattern that
-/// should be revisited once finer-grained dependencies are explored more
-/// thoroughly.
-inline DependencyScope getScopeForAccessLevel(AccessLevel l) {
-  switch (l) {
-  case AccessLevel::Private:
-  case AccessLevel::FilePrivate:
-    return DependencyScope::Private;
-  case AccessLevel::Internal:
-  case AccessLevel::Public:
-  case AccessLevel::Open:
-    return DependencyScope::Cascading;
-  }
-  llvm_unreachable("invalid access level kind");
-}
-
-// A \c DependencySource is currently defined to be a parent source file and
-// an associated dependency scope.
+// A \c DependencySource is currently defined to be a parent source file.
```

Comment:
"parent"? Enclosing lexically? Or lookup parent?

Reply:
Now that we're in pure-private-dependencies-land, the source is always a primary. Going to update this.

Reply:
Thanks!